src/personal.js

```diff
@@ -38,6 +48,24 @@ $(function() {
 		})
 	}
 
+	PersonalSettings.prototype.updateZoteroAPIKey = function(key) {
+		const self = this
+		this._updateSetting({ zoteroAPIKeyInput: key }, function() {
+			self.zoteroAPIKeyInput.value = key
+		}, function() {
+
+		})
```

Comment:
```suggestion
		this._updateSetting({ zoteroAPIKeyInput: key }, () => {
			this.zoteroAPIKeyInput.value = key
		}, (error) => {
			console.error('Failed to update the Zotero API key', error)
		})
```

Would also be great to add a call to https://www.npmjs.com/package/@nextcloud/dialogs `showError` to display a message to the user if this fails.

Reply:
used showError :)